00_ES6/04 redux_saga/src/sagas/index.js

```diff
@@ -0,0 +1,77 @@
+import { messageFactory } from '../api/chat'
+import io from 'socket.io-client';
+import { eventChannel } from 'redux-saga';
+import { fork, take, call, put, cancel } from 'redux-saga/effects';
+import {actionIds} from '../common';
+import {
+  EnrollRoomRequest, disconnectRoomRequest, onDisconnect,
+  onMessageReceived, onMessageListReceived, sendMessage
+} from '../actions';
+import {establishRoomSocketConnection} from './business';
+
+
+function connect(sessionInfo) {
+  const socket = establishRoomSocketConnection(sessionInfo.nickname, sessionInfo.room);
```

Comment:
What if the socket can not get resolved, for example no server running, that will be an unhandle exception, that can make app crashing. Is in the scope of this issue, or maybe in anorther one?

Reply:
In the subscribe method we have this handler:

```
    socket.on('error', (error) => {
      // TODO: handle
    });
```

Would it be handled there? Or we need to make something prior to this?

Reply:
I think that it could be previous to that, we have to test it, because I'm not really sure.

Reply:
Maybe we can live with that (Add some TODO), but it would be good idea to know the right answer, maybe @crsanti that has been working with sockets can shed us some light

Reply:
AFAIK using Phoenix, a "Socket" (PhoenixSocket) instance is created, if by any change you cannot establish a socket communication an error is emited using subscription (like onError) but there's no Error thrown to the main process.

Reply:
What I see with the Braulio's implementation is that if you cannot connect to the socket (bad endpoint/connectivity issues) the socket won't be resolved and main saga will be _stuck_ at waiting for the socket to connect.